tests_fp/test_time_fp.py

```diff
@@ -116,17 +116,67 @@ def test_block_time_init(self):
         assert test_time.time == FixedPoint(0)
         assert test_time.block_number == FixedPoint(0)
         assert test_time.step_size == FixedPoint("1.0") / FixedPoint("365.0")
+        with self.assertRaises(ValueError):
+            test_time = time.BlockTimeFP(time_unit="seconds")
 
     def test_block_time_setters(self):
         """Test attribute setters for block time"""
         test_time = time.BlockTimeFP()
-        # default step_size
+        # time
+        assert test_time.time == FixedPoint(0)
+        test_time.set_time(FixedPoint("4.0"))
```

Comment:
letting set_time accept `str` input and doing this cast to `FixedPoint` inside `set_time` would avoid a lot of boilerplate